x-pack/plugins/fleet/public/index.ts

```diff
@@ -62,9 +64,15 @@ export { PackagePolicyEditorDatastreamPipelines } from './applications/fleet/sec
 export type { PackagePolicyEditorDatastreamPipelinesProps } from './applications/fleet/sections/agent_policy/create_package_policy_page/components/datastream_pipelines';
 export { PackagePolicyEditorDatastreamMappings } from './applications/fleet/sections/agent_policy/create_package_policy_page/components/datastream_mappings';
 export type { PackagePolicyEditorDatastreamMappingsProps } from './applications/fleet/sections/agent_policy/create_package_policy_page/components/datastream_mappings';
-
 export type { DynamicPagePathValues } from './constants';
 
+export type { InputFieldProps } from './applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_var_field';
```

Comment:
don't see the usage in our plugin. Do you still need to export it?

Reply:
@maxcold 
yes, if we don't export it, CI will complain about TS as shown in this build https://buildkite.com/elastic/kibana-pull-request/builds/200909#018ea4d7-4e90-4a1b-b130-44e85b5c45c9

I asked @opauloh  about this weird TS error

Reply:
Yes that error happens when consuming other plugin components that use types in it's declaration, such as this: 

```js
PackagePolicyInputVarField: React.FunctionComponent<InputFieldProps> = memo(
```

When not exporting the type it happens the following error:

```
x-pack/plugins/fleet/public/index.ts:69:14 - error TS4023: Exported variable 'PackagePolicyInputVarField' has or is using name 'InputFieldProps' from external module "kibana/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/steps/components/package_policy_input_var_field" but cannot be named.
```

When I researched that, I found out that this error is caused by referencing locally imported types such as `InputFieldProps`, the compiler is failing to figure out the exact shape of `PackagePolicyInputVarField`, because it does not know the shape of `InputFieldProps`. 

Weirdly it applies only when importing from other plugins, so I'm assuming it's something related to how Kibana `tsconfigs` are configured, didn't dig more into that, but I concluded that exporting the type solves the issue without increasing on bundle size, so it's a good option. 

Maybe adding comment before this export could help explain that behaviour for future folks